test/odbc/src/drivers.h

```diff
@@ -0,0 +1,38 @@
+#ifndef DRIVERS_H
+#define DRIVERS_H
+
+#include "constants.h"
+#include "connection_object.h"
+
+using namespace constants;
+
+class Drivers {
+
+  public:
+
+    static map<ServerType, ConnectionObject> GetOdbcDrivers() {
```

Comment:
Probably no need for singleton, and probably no need for this function.

The Drivers class definition could look something like this:
(note did not try to compile this so it maybe 'sudo code', also the implementations of the functions listed here should be in drivers.cpp file)

```
class Drivers {

  public:

    /*  
    Probably not needed
    static map<ServerType, ConnectionObject> GetOdbcDrivers() {
      if (odbc_drivers_.empty()) {
         SetDrivers();
     }
      return odbc_drivers_;
    }
    */

    static bool DriverExists(ServerType st) {
      if (odbc_drivers_.empty()) {
         SetDrivers();
     }
      return odbc_drivers_.find(st) != odbc_drivers_.end()
    }

    static ConnectionObject &co GetDriver(ServerType st) {
      if (odbc_drivers_.empty()) {
         SetDrivers();
     }
      return odbc_drivers_.(ServerType::PSQL);
    }

    // Constructor
    Drivers();

    // Destructor
    ~Drivers();

    private:
  
    static void SetDrivers();

    static map<ServerType, ConnectionObject> odbc_drivers_;

    // Goes through config.txt and returns a map with values from the configuration file
    static map<string, string> ParseConfigFile();

    // Checks if given connection string parameters forms a valid connection object
    static bool IsValidConnectionObject(string driver, string server, string port, string uid, string pwd, string dbname);
};


```

Given the above calls in the Setup functions would look like this:

```
void SetUp() override {
    
    if(! Drivers::DriverExists(ServerType::PSQL))
    {
      GTEST_SKIP() << "PSQL Driver not present: skipping all tests for this fixture.";
    }

```
and calls to OdbcHandler constructors could look like this:

`OdbcHandler odbcHandler(Drivers::GetDriver(ServerType::PSQL));`

These would hide the internal detail that the Drivers class keeps the drivers in map structure.
And would be easier to read and write.

Reply:
Done.